crates/matrix-sdk/src/client/mod.rs

```diff
@@ -1017,6 +1019,47 @@ impl Client {
         LoginBuilder::new_password(self.clone(), id, password.to_owned())
     }
 
+    /// Login to the server with a custom login type
+    ///
+    /// # Arguments
+    ///
+    /// * `login_type` - Identifier of the custom login type, e.g. `org.matrix.login.jwt`
```

Comment:
Is there an MSC for this? If yes, it would be preferable to add support for it to Ruma and expose a first-party login method for it in the SDK, rather than having a `JsonObject`-based API.

Reply:
Not one known to me. The method I am using is https://matrix-org.github.io/synapse/latest/jwt.html

Reply:
Currently I use a custom http client to call the endpoint, and then `Client::restore_session`. Unfortunately, it requires a custom parsing of the response and it does not use the http client and retry configuration of `matrix_sdk::Client`.

Reply:
I see 👍🏼

Reply:
As per rustfmt error in CI:

```suggestion
    /// * `login_type` - Identifier of the custom login type, e.g.
    ///   `org.matrix.login.jwt`
```